src/script_lua.c

```diff
@@ -1650,23 +1650,6 @@ void luaCallFunction(scriptRunCtx* run_ctx, lua_State *lua, robj** keys, size_t
         err = lua_pcall(lua,2,1,-4);
     }
 
-    /* Call the Lua garbage collector from time to time to avoid a
-     * full cycle performed by Lua, which adds too latency.
-     *
-     * The call is performed every LUA_GC_CYCLE_PERIOD executed commands
-     * (and for LUA_GC_CYCLE_PERIOD collection steps) because calling it
-     * for every command uses too much CPU. */
-    #define LUA_GC_CYCLE_PERIOD 50
-    {
-        static long gc_count = 0;
-
-        gc_count++;
-        if (gc_count == LUA_GC_CYCLE_PERIOD) {
-            lua_gc(lua,LUA_GCSTEP,LUA_GC_CYCLE_PERIOD);
-            gc_count = 0;
-        }
-    }
-
```

Comment:
Why not call `luaGC` here (as it was), this will cover both the function and the eval no?

Reply:
in the original code we share `gc_count` here for script and function, and now they need individual `gc_count` to do GC after loading and executing.
if we didn't separate it into script and function, we would need three `gc_count` to keep track.

Reply:
Is it that bad not to separate?
I am not follow why we would need 3? Is it for loading eval and functions? Whats the reason not to just merge them all with a single counter?

Reply:
Right better to separate